core/trino-main/src/main/java/io/trino/connector/WorkerDynamicCatalogManager.java

```diff
@@ -28,8 +29,11 @@
 import java.util.List;
 import java.util.Map.Entry;
 import java.util.Set;
+import java.util.concurrent.CompletableFuture;
```

Comment:
This PR is using both JVM CompletableFutures and Guava ListenableFutures.  Pick one or the others.  Generally we try to avoid JVM CompletableFutures because they have non-intuitive behaviors around error handling and canceling.